### PR TITLE
feat: add dependency track integration

### PR DESCRIPTION
**Pull request by takumakume**
ref https://github.com/aquasecurity/postee/issues/575

SBOM Report has been supported since Trivy Operator [0.15.0](https://github.com/aquasecurity/trivy-operator/releases/tag/v0.15.0) .

I want to execute a webhook to Dependency Track via Postee. (ref: https://github.com/aquasecurity/trivy-operator/issues/143#issuecomment-1582164314 )

In this Pull Request, I implemented the following:

- Added Integration of Dependency Track
- Added Template for sending Trivy Operator sbomReport to Dependency Track

Dependency Track uploads the BOM with ProjectName and ProjectVersion as keys.

Dependency Track Integration has the following specification:

- `ProjectName:ProjectVersion` is the template `title` .  (e.g. `busybox:latest`)
- BOM data is in the template `description` .

In this Pull Request, only JSON used by trivy-operator is supported as BOM format.

---

The operation was checked in the following environment.

- Dependency Track 4.8.0
- Trivy Operator 0.15.**1** (ref: https://github.com/aquasecurity/trivy-operator/pull/1378)

```
routes:
  - name: dependencytrack
    input: contains(input.kind, "SbomReport")
    actions: [ dependencytrack ]
    template: dependencytrack
actions:
  - name: my-dependencytrack
    type: dependencytrack
    enable: true
    url: http://192.168.100.28:8081/ # local docker compose
    dependency-track-api-key: **
templates:
  - name: dependencytrack
    rego-package: postee.trivyoperator.dependencytrack
```

- trivy-operator sends SBOM Report webhook to Postee and is registered in Dependency Track.
- Can be configured from Postee UI
  image

### CODE REVIEW

**Comment by takumakume** [comment]
https://aquasecurity.slack.com/archives/C02NT2Y4FJL/p1691406606182759

> Hi all, I would just quickly like to remind everyone that our efforts have shifted to Trivy and Tracee, you are still welcome to use Postee -- however, it is not actively being developed upon by the core team

Will this pull request be reviewed and merged?

File: cfg.yaml [hunk 1]
```diff
@@ -141,6 +143,12 @@ actions:
   url:              # Mandatory. Url of Nexus IQ server
   organization-id:  # Mandatory. Organization UID like "222de33e8005408a844c12eab952c9b0"
 
+  - name: my-dependencytrack
+    type: dependencytrack
+    enable: false
+    url:                       # Mandatory. Url of Dependency Track server
```
Comment (DmitriyLewen) [inline cfg.yaml:149]: can we add some example?
e.g. from docker deploying - `http://localhost:8080/api`

File: actions/dependencytrack.go [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+		ProjectVersion: projectAndVersion[1],
+		AutoCreate:     true,
+		BOM:            base64.StdEncoding.EncodeToString(bom),
+	})
+
+	if err != nil {
+		return fmt.Errorf("failed to upload BOM: %w", err)
+	}
+
+	return nil
```
Comment (DmitriyLewen) [inline actions/dependencytrack.go:65]: Can we add a message that the submission is complete?

**Review by DmitriyLewen — COMMENTED** [review]
Hello @takumakume 
Sorry for waiting!

I left some comments.

**Comment by takumakume** [comment]
@DmitriyLewen 
Thanks for the check! I have push some fixes.

**Review by DmitriyLewen — APPROVED** [review]
Thanks for your work!
LGTM.
@simar7 can you take a look and merge PR?

**Comment by takumakume** [comment]
Hi, is there anything I can do to merge this pull request? Thank you for your time. @simar7

**Pull request merged**